clouddrift/adapters/gdp1h.py

```diff
@@ -78,46 +89,59 @@ def download(
 
     # adjust the tmp_path if using the experimental source
     if tmp_path is None:
-        tmp_path = GDP_TMP_PATH if url == GDP_DATA_URL else GDP_TMP_PATH_EXPERIMENTAL
+        tmp_path = (
+            GDP_TMP_PATH
+            if url == GdpDataUrl.GDP_DATA_URL_STABLE
+            else GDP_TMP_PATH_EXPERIMENTAL
+        )
 
-    print(f"Downloading GDP hourly data from {url} to {tmp_path}...")
+    _logger.debug(f"Downloading GDP hourly data from ({url}) to ({tmp_path})")
 
     # Create a temporary directory if doesn't already exists.
     os.makedirs(tmp_path, exist_ok=True)
 
-    if url == GDP_DATA_URL:
+    if url == GdpDataUrl.GDP_DATA_URL_STABLE:
         pattern = "drifter_hourly_[0-9]*.nc"
         filename_pattern = "drifter_hourly_{id}.nc"
-    elif url == GDP_DATA_URL_EXPERIMENTAL:
+    elif url == GdpDataUrl.GDP_DATA_URL_EXPERIMENTAL:
         pattern = "drifter_hourly_[0-9]*.nc"
         filename_pattern = "drifter_hourly_{id}.nc"
+    else:
+        raise ValueError(f"{url} is not a valid ")
 
     # retrieve all drifter ID numbers
+    chosen_drifter_ids: ArrayLike
     if drifter_ids is None:
         urlpath = urllib.request.urlopen(url)
         string = urlpath.read().decode("utf-8")
-        filelist = re.compile(pattern).findall(string)
-        drifter_ids = np.unique([int(f.split("_")[-1][:-3]) for f in filelist])
+        filelist: List[str] = re.compile(pattern).findall(string)
+        chosen_drifter_ids = np.unique([int(f.split("_")[-1][:-3]) for f in filelist])
+    else:
+        chosen_drifter_ids = drifter_ids
 
     # retrieve only a subset of n_random_id trajectories
     if n_random_id:
-        if n_random_id > len(drifter_ids):
-            warnings.warn(
-                f"Retrieving all listed trajectories because {n_random_id} is larger than the {len(drifter_ids)} listed trajectories."
+        if n_random_id > len(chosen_drifter_ids):
+            _logger.warn(
+                f"Retrieving all listed trajectories because {n_random_id} is larger than the {len(chosen_drifter_ids)} listed trajectories."
             )
         else:
             rng = np.random.RandomState(42)
-            drifter_ids = sorted(rng.choice(drifter_ids, n_random_id, replace=False))
+            drifter_ids = sorted(
+                rng.choice(chosen_drifter_ids, n_random_id, replace=False)
+            )
```

Comment:
I think this should be `chosen_drifter_ids` since this is returned by that function.

```suggestion
            chosen_drifter_ids = sorted(
                rng.choice(chosen_drifter_ids, n_random_id, replace=False)
            )
```